### PR TITLE
[Admin] Added variant autocomplete to the forms of product images

### PR DESCRIPTION
**Pull request by NeverResponse**
| Q               | A
| --------------- | ---
| Bug fix?        | no |
| New feature?    | yes |
| BC breaks?      | yes |
| Related tickets | - |
| License         | MIT |

### CODE REVIEW

File: features/product/managing_product_variants/autocomplete_product_variant_descriptor.feature [hunk 1]
```diff
@@ -0,0 +1,37 @@
+@managing_product_variants
+Feature: Product variants autocomplete
+    In order to get hints when looking for named product variants
+    As an Administrator
+    I want to get product variants according to my specified phrase
+
+    Background:
+        Given the store operates on a single channel in "United States"
+        And the store has a "Snake" configurable product
+        And this product has "Ouroboros", "Boomslang" and "Bimini" variants
```
Comment (pamil) [inline features/product/managing_product_variants/autocomplete_product_variant_descriptor.feature]: Ouroboros 👍

File: src/Sylius/Behat/Context/Api/Admin/ManagingProductVariantsContext.php [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    public function __construct(Client $client, SessionInterface $session)
+    {
+        $this->client = $client;
+        $this->session = $session;
+    }
+
+    /**
+     * @When I look for a variant with :phrase in descriptor within the :product product
+     */
+    public function iTypeIn($phrase, ProductInterface $product)
```
Comment (pamil) [inline src/Sylius/Behat/Context/Api/Admin/ManagingProductVariantsContext.php]: `iTypeIn` -> `iLookForVariantWithDescriptorAndProduct` (or sth like that)?

File: src/Sylius/Behat/Context/Api/Admin/ManagingProductVariantsContext.php [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    {
+        $this->client->getCookieJar()->set(new Cookie($this->session->getName(), $this->session->getId()));
+        $this->client->request(
+            'GET',
+            '/admin/ajax/product-variants/search',
+            ['phrase' => $phrase, 'productCode' => $product->getCode()],
+            [],
+            ['ACCEPT' => 'application/json']
+        );
```
Comment (pamil) [inline src/Sylius/Behat/Context/Api/Admin/ManagingProductVariantsContext.php]: I'm thinking about making interaction with this endpoint a page object, wdyt?

File: src/Sylius/Behat/Context/Api/Admin/ManagingTaxonsContext.php [hunk 4]
```diff
@@ -86,7 +86,7 @@ public function iShouldSeeTaxonsInTheList($number)
     {
         $response = json_decode($this->client->getResponse()->getContent(), true);
 
-        Assert::eq($number, count($response));
+        Assert::eq(count($response), $number);
```
Comment (pamil) [inline src/Sylius/Behat/Context/Api/Admin/ManagingTaxonsContext.php]: 👍

File: src/Sylius/Behat/Context/Setup/ProductContext.php [hunk 5]
```diff
@@ -335,6 +335,50 @@ public function theProductHasVariantPricedAt(
     }
 
     /**
+     * @Given /^the (product "[^"]+") has(?:| a| an) "([^"]+)" variant$/
+     * @Given /^(this product) has(?:| a| an) "([^"]+)" variant$/
+     * @Given /^(this product) has "([^"]+)", "([^"]+)" and "([^"]+)" variants$/
+     */
+    public function theProductHasVariants(ProductInterface $product, ...$variants)
```
Comment (pamil) [inline src/Sylius/Behat/Context/Setup/ProductContext.php]: `$variants` -> `$variantsNames`?

**Comment by pjedrzejewski** [comment]
@NeverResponse Requires rebasing. :)

**Comment by pjedrzejewski** [comment]
Thanks Jasiek! 👍

**Pull request merged**